### DISCUSSION TITLE
Using non standard header for bearer token

### DISCUSSION THREAD

**Discussion by avanide** [grafana/grafana-infinity-datasource#1083] [Q&A]
Hi,
We have an internal system that is not following the specification and is using another header for the bearer token:

- It should be the header "Authorization: bearer <accesstoken>"
- Our system is using the header "X-Access-Token: <accesstoken>".
This system is using Grafana and the infinity datasource. Grafana is receiving this header "X-Access-Token". The datasource (rest apis) needs to receive the value of this header.

Is there a way to forward our non standard header from Grafana to the datasource? (X-Access-Token)? In the documentation (https://grafana.com/docs/plugins/yesoreyeram-infinity-datasource/latest/setup/authentication/#bearer-token), there's a TIP but it's unclear how to use it: "If you need a custom prefix instead of Bearer prefix, use API key authentication instead with the key of Authorization."

Alternatively, we have considered to use client credentials so that Grafana is able to get its own access token that is then sent to the datasource. When we use that, we did not find how to specify another header for the access token. Is there a way to do it?

**Comment by yesoreyeram** [comment]
If they are static credentials, you can add it like any other header. If that is oauth type and need refresh, follow  https://github.com/grafana/grafana-infinity-datasource/pull/868 for updates
